cpp/src/ray/runtime/abstract_ray_runtime.cc

```diff
@@ -87,8 +87,8 @@ void AbstractRayRuntime::Put(std::shared_ptr<msgpack::sbuffer> data,
 }
 
 std::string AbstractRayRuntime::Put(std::shared_ptr<msgpack::sbuffer> data) {
-  ObjectID object_id =
-      ObjectID::FromIndex(worker_->GetCurrentTaskID(), worker_->GetNextPutIndex());
+  ObjectID object_id = ObjectID::FromIndex(GetWorkerContext().GetCurrentTaskID(),
```

Comment:
Actually, I'm saying that this should be 
```
ObjectID object_id
````
and overriden to instantiate objectID only for local mode etc. Pls refer to `CoreWorker::Put` definition https://github.com/ray-project/ray/blob/aa35045b6f701efa148489e30d0a36728de52000/src/ray/core_worker/core_worker.cc#L798

Reply:
So, you don't even need the virtual method for native mode.

Reply:
Yes, this is another way, and i will use it, local mode override the put is ok.
 The virtual GetWorkerContext is needed, because the member worker_ should not belong to the abstract ray runtime, it only belongs to local mode, and local mode submitter need to get the context, so need provide a virtual GetWorkerContext for it.